src/solver/infeasible-problem-analysis/problem.h

```diff
@@ -11,22 +11,77 @@ namespace Antares
 {
 namespace Optimization
 {
-class InfeasibleProblemAnalysis
+
+
+class SingleAnalysis
 {
 public:
-    InfeasibleProblemAnalysis() = delete;
-    explicit InfeasibleProblemAnalysis(const std::string& solverName, const PROBLEME_SIMPLEXE_NOMME* ProbSpx);
-    InfeasibleProblemReport produceReport();
+    SingleAnalysis(std::shared_ptr<operations_research::MPSolver> problem_);
+    virtual void run() = 0;
+    virtual void printReport() = 0;
+    virtual std::string title() = 0;
+    bool hasDetectedInfeasibilityCause() { return hasDetectedInfeasibilityCause_; }
+
+protected:
+    std::shared_ptr<operations_research::MPSolver> problem_;
+    bool hasDetectedInfeasibilityCause_ = false;
+};
+
+
+class SlackVariablesAnalysis : public SingleAnalysis
+{
+    using SingleAnalysis::SingleAnalysis;
+public:
+    void run() override;
+    void printReport() override;
+    std::string title() override { return "Slack variables analysis"; }
```

Comment:
This interface class isn't really needed

Reply:
See answer above